src/app.ts

```diff
@@ -10,6 +10,7 @@ export default async function serviceApp(
   fastify: FastifyInstance,
   opts: FastifyPluginOptions
 ) {
+  delete opts.skipOverride // This option only serves testing purpose
```

Comment:
As I feared, I think we've introduced a problem here: https://github.com/fastify/fastify-cli/pull/742/files#r1689541348

For example, `@fastify/mysql` [passes all the given options directly to `mysql2`](https://github.com/fastify/fastify-mysql/blob/10a2a8e34911d1982ad3556b03424c276e7ccdcf/index.js#L62), which triggers warnings from `mysql2`:
```
Ignoring invalid configuration option passed to Connection: skipOverride. This is currently a warning, but in future versions of MySQL2, an error
will be thrown if you pass an invalid configuration option to a Connection
```

Maybe it's also a bad design to propagate all the options in the autoloader? Like the cli generated template suggest to do:
```ts
export default async function serviceApp(
  fastify: FastifyInstance,
  opts: FastifyPluginOptions
) {
  await fastify.register(fastifyAutoload, {
      dir: path.join(import.meta.dirname, "plugins"),
      options: { ...opts } // opts propagated here
  });
  
  // code...
}
```

Reply:
Not sure what to say here, but you are right. Passing all default options that results into this error, is NOT good.